### ISSUE TITLE
java.nio.file.FileSystemNotFoundException on FileSystems.getFileSystem

### ISSUE THREAD

**Issue by stefanofornari** [carlspring/s3fs-nio#716]
# Bug Description

` FileSystems.getFileSystem()` throws a `FileSystemNotFoundException` if the file system has not been created before. This could be acceptable if `FileSystems.newFileSystem(URI.create(file), new HashMap<>())` would be smart enough to return an existing FS if already created. Instead `newFileSystem()` throws a `FileSystemAlreadyExistsException` if called twice. 
Alternatively, getFileSystem could return null, so to allow the pattern below (which now does not work):

```
FileSystem fs = FileSystems.getFileSystem(URI.create(file));
if (fs == null) {
        fs = FileSystems.newFileSystem(URI.create(file), new HashMap<>());
}

```

# Steps To Reproduce

see description

# Expected Behavior

* getFileSystem() returns a new FileSystem if not already created?
* getFileSystem() returns null?
* newFileSystem() returns an existing FS if already created?

# Environment

* `s3fs-nio` version: `1.0.0` (also `1.0.1`)

**Comment by steve-todorov** [comment]
Hey @stefanofornari ,

Thanks for reporting this.

There is a bug left in some of the legacy code we've inherited. It was preventing an internal property's state to be visible from other threads and this messed up the `getFileSystem` method logic. I believe this should be fixed with #717. Could you try it out and let me know?

Your suggestion for returning `null` is not possible because it contradicts the `java.nio.file.FileSystems#getFileSystem` method's documentation:

![image](https://github.com/carlspring/s3fs-nio/assets/2303663/6e2c0e69-e77d-4b38-a8fb-e1c1b6cae5a5)

Furthermore, the `FileSystems` method is a `final` one so we can't really change anything.
You can wrap your code in a `try/catch` that catches `java.nio.file.ProviderNotFoundException` instead of doing a `null` check and you'll achieve the same result :)

**Comment by steve-todorov** [comment]
My bad then. Anyway let me know if this fixes it :)
